controllers/reconcilers/kserve_authconfig_reconciler.go

```diff
@@ -0,0 +1,108 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconcilers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	kservev1beta1 "github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+	authorinov1beta2 "github.com/kuadrant/authorino/api/v1beta2"
+	"github.com/opendatahub-io/odh-model-controller/controllers/resources"
+	"github.com/pkg/errors"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type KserveAuthConfigReconciler struct {
+	client         client.Client
+	scheme         *runtime.Scheme
+	detector       resources.AuthTypeDetector
+	store          resources.AuthConfigStore
+	templateLoader resources.AuthConfigTemplateLoader
+}
+
+func NewKserveAuthConfigReconciler(client client.Client, scheme *runtime.Scheme) *KserveAuthConfigReconciler {
+	return &KserveAuthConfigReconciler{
+		client:         client,
+		scheme:         scheme,
+		detector:       resources.NewKServeAuthTypeDetector(client),
+		store:          resources.NewClientAuthConfigStore(client),
+		templateLoader: resources.NewConfigMapTempalteLoader(client, resources.NewStaticTemplateLoader()),
+	}
+}
+
+func (r *KserveAuthConfigReconciler) Reconcile(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) error {
+
+	if isvc.Status.URL == nil {
+		log.V(1).Info("Inference Service not ready yet, waiting for URL")
```

Comment:
imho, it should be a warning.

Reply:
Could make it a warning. It's part of the normal flow to reconcile a few times before the Status.URL show up

Reply:
So, `log` does not have a warning func. Only `Info` and `Error`.
We have `V(1)` in the middle to raise verbosity, so I think this is good enough.